libs/eavmlib/src/ahttp_client.erl

```diff
@@ -0,0 +1,411 @@
+
+-opaque connection() :: #http_client{}.
+
+-export_type([connection/0, error_tuple/0, backend/0]).
+
+%%-----------------------------------------------------------------------------
+%% @param   Protocol the protocol, either http or https
+%% @param   Host the server hostname
+%% @param   Port the server port number, usually 80 (http) or 443 (https)
+%% @param   Options the property list with http client and connection options
+%% @returns Either a http connection result tuple or an error tuple
+%% @doc     Connects to the http(s) server.
+%%
+%%          The `Host' parameter may be a fully qualified host name or a string
+%%          containing a valid dotted pair IP address.  (Currently, only IPv4 is
+%%          supported). `Host' can be also a binary.
+%%
+%%          The `Options' can be used for providing connection options such as SSL
+%%          `{verify, verify_none}', `gen_tcp' options such as `{active, false}' and `ahttp_client'
+%%          `{parse_headers, [<<"HeaderName">>]}'.
+%% @end
+%%-----------------------------------------------------------------------------
+-spec connect(
+    Protocol :: protocol(), Host :: host(), Port :: inet:port_number(), Options :: [option()]
+) -> {ok, connection()} | error_tuple().
+
+connect(Protocol, Host, Port, Options) when is_binary(Host) ->
+    connect(Protocol, binary_to_list(Host), Port, Options);
+connect(http, Host, Port, Options) ->
+    {ParseHeaders, ConnOptions} = take_parse_headers(Options),
+    case gen_tcp:connect(Host, Port, transform_options(ConnOptions)) of
+        {ok, Sock} -> init(Host, {gen_tcp, Sock}, ParseHeaders);
+        {error, Error} -> {error, {gen_tcp, Error}}
+    end;
+connect(https, Host, Port, Options) ->
+    {ParseHeaders, ConnOptions} = take_parse_headers(Options),
+    case ssl:connect(Host, Port, transform_options(ConnOptions)) of
+        {ok, Sock} -> init(Host, {ssl, Sock}, ParseHeaders);
+        {error, Error} -> {error, {ssl, Error}}
+    end.
+
+init(Host, TSocket, ParseHeaders) ->
+    {ok, #http_client{
+        host = iolist_to_binary(Host), socket = TSocket, parse_headers = ParseHeaders
+    }}.
+
+transform_options(Options) ->
+    WithActive =
+        case proplists:get_value(active, Options) of
+            undefined -> [{active, true} | Options];
+            _OtherValue -> Options
+        end,
+    [{binary, true} | WithActive].
+
+take_parse_headers(Options) ->
+    case proplists:get_value(parse_headers, Options) of
+        undefined -> {[], Options};
+        Value -> {Value, proplists:delete(parse_headers, Options)}
+    end.
+
+%%-----------------------------------------------------------------------------
+%% @param   Conn the connection
+%% @param   Method a http method such as "GET", "POST", "PUT", etc...
+%% @param   Path the path to the http resource, such as "/"
+%% @param   Headers a list of headers
+%% @param   Body the body that is sent to the server, may be undefined or nil when there is no body
```

Comment:
Also whether it is `nil` or `none` the exact term should probably be code syntax highlighted with an Erlang (`') code block in the `@param` definition.

Reply:
Fixed.